python/tests/res/enkf/test_ensemble_config.py

```diff
@@ -1,17 +1,136 @@
 from tests import ResTest
-from res.test import ErtTestContext
+# from res.test import ErtTestContext
 
-from ecl.util.util import BoolVector,IntVector
-from res.enkf import ActiveMode, EnsembleConfig
-from res.enkf import ObsVector , LocalObsdata
+from res.enkf import EnsembleConfig, ResConfig
+# from res.enkf import ObsVector , LocalObsdata
+from res.enkf import ConfigKeys
+from ecl.util.test import TestAreaContext
+from res.enkf.enums import GenDataFileType
 
 
 class EnsembleConfigTest(ResTest):
 
+    def setUp(self):
+        self.case_directory = self.createTestPath("local/simple_config/")
+        self.case_file = 'simple_config/ensemble_config'
+
     def test_create(self):
         conf = EnsembleConfig( )
         self.assertEqual( len(conf) , 0 )
         self.assertFalse( "XYZ" in conf )
 
         with self.assertRaises(KeyError):
             node = conf["KEY"]
+
+    def test_ensemble_config_constructor(self):
+        config_dict = {
+            ConfigKeys.GEN_KW_TAG_FORMAT: '<%s>',
+            ConfigKeys.GEN_PARAM: [
+                {
+                    ConfigKeys.NAME: 'GP',
+                    ConfigKeys.FORWARD_INIT: False,
+                    ConfigKeys.INPUT_FORMAT: GenDataFileType.ASCII,
+                    ConfigKeys.OUTPUT_FORMAT: GenDataFileType.ASCII,
+                    ConfigKeys.INIT_FILES: 'GP/GP.txt',
+                    ConfigKeys.ECL_FILE: 'GP.txt',
+                    ConfigKeys.MIN_STD: None,
+                    ConfigKeys.TEMPLATE: None,
+                    ConfigKeys.KEY_KEY: None
+                },
+            ],
+            ConfigKeys.GEN_DATA: [
+                {
+                    ConfigKeys.NAME: 'SNAKE_OIL_OPR_DIFF',
+                    ConfigKeys.INPUT_FORMAT: GenDataFileType.ASCII,
+                    ConfigKeys.RESULT_FILE: 'snake_oil_opr_diff_%d.txt',
+                    ConfigKeys.REPORT_STEPS: [199],
+                    ConfigKeys.INIT_FILES: None,
+                    ConfigKeys.ECL_FILE: None,
+                    ConfigKeys.TEMPLATE: None,
+                    ConfigKeys.KEY_KEY: None
+                },
+                {
+                    ConfigKeys.NAME: 'SNAKE_OIL_GPR_DIFF',
+                    ConfigKeys.INPUT_FORMAT: GenDataFileType.ASCII,
+                    ConfigKeys.RESULT_FILE: 'snake_oil_gpr_diff_%d.txt',
+                    ConfigKeys.REPORT_STEPS: [199],
+                    ConfigKeys.INIT_FILES: None,
+                    ConfigKeys.ECL_FILE: None,
+                    ConfigKeys.TEMPLATE: None,
+                    ConfigKeys.KEY_KEY: None
+                }
+            ],
+            ConfigKeys.CUSTOM_KW: [
+                {
+                    ConfigKeys.NAME: 'SNAKE_OIL_NPV',
+                    # ConfigKeys.RESULT_FILE: 'configuration_tests/snake_oil_npv.txt',
```

Comment:
remove commented line

Reply:
We need to discuss this (with @markusdregi ) too I think

Reply:
Please elaborate @xjules and I'll try to pitch in ;)

Reply:
This's been solved in the end, the question I had relates mainly to SCHEDULE_PREDICTION_FILE which contains optional parameters (INIT_FILES, MIN_STD, PARAMETER_KEY) that are not documented nor no utilization has been found by me at least.

Reply:
> no utilization has been found by me at least.

Seems like they are used as parameters to a `GEN_KW` instantiation in ensemble config...

Reply:
Yes, that's true. I was more wondering how they serve SCHEDULE_PREDICTION_FILE, but anyway they can be setup if need